src/main/java/org/opentripplanner/middleware/controllers/api/MonitorTripController.java

```diff
@@ -0,0 +1,57 @@
+package org.opentripplanner.middleware.controllers.api;
+
+import com.mongodb.client.model.Filters;
+import org.bson.conversions.Bson;
+import org.eclipse.jetty.http.HttpStatus;
+import org.opentripplanner.middleware.models.MonitoredTrip;
+import org.opentripplanner.middleware.persistence.Persistence;
+import spark.Request;
+
+import static com.mongodb.client.model.Filters.eq;
+import static org.opentripplanner.middleware.auth.Auth0Connection.isAuthorized;
+import static org.opentripplanner.middleware.utils.JsonUtils.logMessageAndHalt;
+
+/**
+ * Implementation of the {@link ApiController} abstract class for managing monitored trips. This controller connects
+ * with Auth0 services using the hooks provided by {@link ApiController}.
+ */
+public class MonitorTripController extends ApiController<MonitoredTrip> {
+    private static final long MAXIMUM_ALLOWED_MONITORED_TRIPS = 5;
+
+    public MonitorTripController(String apiPrefix) {
+        super(apiPrefix, Persistence.monitoredTrip, "secure/monitortrip");
+    }
+
+    @Override
+    MonitoredTrip preCreateHook(MonitoredTrip monitoredTrip, Request req) {
+        isAuthorized(monitoredTrip.userId, req);
+        reachedMaximum(monitoredTrip.userId, req);
+
+        return monitoredTrip;
+    }
+
+    @Override
+    MonitoredTrip preUpdateHook(MonitoredTrip monitoredTrip, MonitoredTrip preExisting, Request req) {
+        isAuthorized(monitoredTrip.userId, req);
+        return monitoredTrip;
+    }
+
+    @Override
+    boolean preDeleteHook(MonitoredTrip monitoredTrip, Request req) {
+        // Authorization checks are done prior to this hook
+        return true;
+    }
+
+    /**
+     * Confirm that the maximum number of saved monitored trips has not been reached
+     */
+    private void reachedMaximum(String userId, Request request) {
```

Comment:
Can we rename this to `verifyBelowMaxNumTrips`?

Reply:
Updated.